packages/test/test-version-utils/src/versionUtils.ts

```diff
@@ -589,3 +589,60 @@ export function versionHasMovedSparsedMatrix(version: string): boolean {
 		version >= "2.0.0-internal.2.0.0" || (!version.includes("internal") && version >= "2.0.0")
 	);
 }
+
+/**
+ * @internal
+ */
+export function versionToComparisonNumber(version: string): number {
+	if (version.startsWith("0.")) {
+		return 0;
+	}
+	if (version.startsWith("1.")) {
+		return 1;
+	}
+	if (version.startsWith("2.0.0-internal.1")) {
+		return 2;
+	}
```

Comment:
You're basically building a map of starting strings --> indices. So you could just have an array of strings, and the value you return is the index of the first matching startsWith in the array.

```ts
const versionOrder["0.", "1." ...];
return versionOrder.findIndex((prefix) => version.startsWith(prefix))
```